python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -257,6 +261,11 @@ def max_task_output_bytes_to_read(self, op: PhysicalOperator) -> Optional[int]:
         """Return the maximum bytes of pending task outputs can be read for
         the given operator. None means no limit."""
         ...
+    
+    @abstractmethod
+    def update_stats_metrics(self, op: PhysicalOperator) -> Optional[int]:
+        """Updates operator metrics"""
+        ...
```

Comment:
perhaps `raise NotImplemented` is more clear?

Reply:
could we rename this to something like `update_op_runtime_metrics()`, to differentiate from the similarly named `StreamingExecutor._update_stats_metrics()`?

Reply:
also, i think the `op` parameter is missing in the implementation of `ReservationOpResourceAllocator`, which subclasses `OpResourceAllocator`. should we remove if from this abstract method?